### PR TITLE
Add MSE

### LINKED ISSUE #43: implement MSE
We need Mean Squared Error!

### PR DESCRIPTION
**Pull request by Bartzi**
Adds mean squared error function :tada: and fixes #43

### CODE REVIEW

File: length/tests/function_tests/test_mean_squared_error.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            [1, 0, 0, 1],
+            [0, 0, 0, 1],
+            [0, 0, 0, 0]
+        ])
+
+    return data, data_2
+
+
+def test_mean_squared_error_forward_zero_loss():
+    data = Graph(np.array([0, 0, 0, 1], dtype=constants.DTYPE))
```
Comment (Jopyth) [inline length/tests/function_tests/test_mean_squared_error.py]: `init()`?

File: length/tests/function_tests/test_mean_squared_error.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    label = Graph(np.array([0, 0, 0, 1], dtype=constants.DTYPE))
```
Comment (Jopyth) [inline length/tests/function_tests/test_mean_squared_error.py]: `init()`?

Comment (Jopyth) [inline length/tests/function_tests/test_mean_squared_error.py]: `init()`?

**Comment by Bartzi** [comment]
please review again =)

File: length/tests/function_tests/test_mean_squared_error.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        (gradients,)
+    )
+
+    gradient_checker.assert_allclose(computed_gradient_1, numerical_gradient_1)
+    gradient_checker.assert_allclose(computed_gradient_2, numerical_gradient_2)
+
+
+def test_mean_squared_error_backward_with_label():
+    data, data_2 = fixed_case(with_label=True)
+    gradients = np.array([2], dtype=constants.DTYPE)
```
Comment (Jopyth) [inline length/tests/function_tests/test_mean_squared_error.py]: same here

File: length/tests/function_tests/test_mean_squared_error.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+def test_mean_squared_error_forward_int_input():
+    data, labels = fixed_case(with_label=True)
+
+    mse = mean_squared_error(Graph(data), Graph(labels))
+    assert math.isclose(float(mse.data), 0.583, abs_tol=1e-3)
+
+
+def test_mean_squared_error_backward():
+    data, data_2 = fixed_case()
+    gradients = np.array([2], dtype=constants.DTYPE)
```
Comment (Jopyth) [inline length/tests/function_tests/test_mean_squared_error.py]: same here

**Review by Jopyth — APPROVED** [review]

**Pull request merged**